lib/compress/zstd_double_fast.c

```diff
@@ -154,7 +154,7 @@ size_t ZSTD_compressBlock_doubleFast_generic(
 
         /* check noDict repcode */
         if ( dictMode == ZSTD_noDict
-          && ((offset_1 > 0) & (MEM_read32(ip+1-offset_1) == MEM_read32(ip+1)))) {
+          && ((offset_1 > 0) && (MEM_read32(ip+1-offset_1) == MEM_read32(ip+1)))) {
```

Comment:
Why is this beneficial? We specifically use `&` so these checks can be run in parallel. By switching to `&&` we force the `offset_1 > 0` to be evaluated first.

If this is very important for performance, could we put the `MEM_read32()` first, and the `offset_1 > 0` second, since the compiler should be able to realize that `offset_1 > 0` doesn't have side effects.

Reply:
Switching the order of the comparisons does very badly with gcc-9 on x86, so I don't recommend that.

Reply:
> Why is this beneficial? We specifically use `&` so these checks can be run in parallel. By switching to `&&` we force the `offset_1 > 0` to be evaluated first.
> 
> If this is very important for performance, could we put the `MEM_read32()` first, and the `offset_1 > 0` second, since the compiler should be able to realize that `offset_1 > 0` doesn't have side effects.

Thanks. I try to test it.